wlauto/workloads/uxperf_applaunch/__init__.py

```diff
@@ -0,0 +1,227 @@
+        Parameter('workload_name', kind=str,
+                  description='Name of the application to launch',
+                  mandatory=True),
+        Parameter('applaunch_type', kind=str, default='launch_from_background',
+                  description="""
+                  Choose launch_from_long-idle for measuring launch time
+                  from long-idle
+                  """),
+        Parameter('applaunch_iterations', kind=int, default=1,
+                  description="""
+                  Number of iterations of the application launch
+                  """),
+    ]
+
+    def __init__(self, device, **kwargs):
+        super(UxperfApplaunch, self).__init__(device, **kwargs)
+        loader = ExtensionLoader()
+        self.workload = loader.get_workload(self.workload_name, device,
+                                            **self.workload_params)
+
+    def validate(self):
+        self.workload.validate()
+        if self.applaunch_type != 'launch_from_long-idle' and \
+                self.applaunch_type != 'launch_from_background':
+            raise ConfigError("Applaunch type specified wrong")
+        self.workload.uiauto_params['package'] = self.workload.package
+        if self.workload.activity:
+            self.workload.uiauto_params[
+                'launch_activity'] = self.workload.activity
+        else:
+            self.workload.uiauto_params['launch_activity'] = "None"
+        self.workload.uiauto_params[
+            'markers_enabled'] = self.workload.markers_enabled
+        self.workload.uiauto_params['applaunch_type'] = self.applaunch_type
+
+    def init_resources(self, context):
+        self.workload.init_resources(context)
+
+    def setup(self, context):
+        AndroidBenchmark.setup(self.workload, context)
+        if self.workload.launch_main is False:  # Required for launching skype
+            self.workload.launch_app()
+
+    def stop_application(self):
+        """kills the application process
+        """
+        self.device.execute('am force-stop {}'.format(self.workload.package))
+
+    def drop_pagecache(self):
+        """clears page cache
+        """
+        self.device.execute('sync')
+        self.device.execute('echo 1 > /proc/sys/vm/drop_caches', as_root=True)
+
+    def drop_inodecache(self):
+        """clears inode cache
+        """
+        self.device.execute('sync')
+        self.device.execute('echo 2 > /proc/sys/vm/drop_caches', as_root=True)
+
+    def drop_inodepagecache(self):
+        """clears inode and page caches
+        """
+        self.device.execute('sync')
+        self.device.execute('echo 3 > /proc/sys/vm/drop_caches', as_root=True)
+
+    def clean_application(self):
+        """Kills the application and clears caches if required
+        """
+        self.stop_application()
+        if(self.applaunch_type == 'launch_from_long-idle'):
+            self.drop_inodepagecache()
+
+    def update_iteration_result(self, context, i):
```

Comment:
To save having to find each log again in `update_results` store a list internally when creating each directory which can then be iterated over.